### ISSUE TITLE
[docfix] Parameter size_average=False behaves differently for mse_loss and cross_entropy

### ISSUE THREAD

**Issue by tarvaina** [pytorch/pytorch#3322]
Setting `size_average=False` behaves differently for `mse_loss` than for `cross_entropy`. For `cross_entropy`, it multiplies the result by the batch size. For `mse_loss` it multiplies the result by the overall size of the matrix.

I believe the `cross_entropy` behavior is correct and `mse_loss` should be changed to behave similarly. Alternatively, the documentation should be changed to clarify the difference.

Actually, `mse_loss` behaves similarly to `kl_div`, but at least the behavior is documented in `kl_div`. (But before reading the documentation, I expected also `kl_div` to behave similarly to `cross_entropy`.) I haven't checked how other loss functions behave. 

```python
import torch.version
from torch.nn import functional as F
from torch import FloatTensor
from torch.autograd import Variable

def v(array):
    return Variable(FloatTensor(array))

def kl(logit, target, size_average=True):
    return F.kl_div(F.log_softmax(logit), target, size_average=size_average)

def xe(logit, target, size_average=True):
    return F.cross_entropy(logit, target.max(1)[1], size_average=size_average)

def mse(logit, target, size_average=True):
    return F.mse_loss(F.softmax(logit), target, size_average=size_average)

target = v(np.eye(5)[:3])
logit = v(np.zeros((3, 5)))

print("Torch version:", torch.__version__)
print()
print("How each loss function scales the result with size_average:")
for loss in [kl, mse, xe]:
    with_size_average = loss(logit, target, size_average=True)
    without_size_average = loss(logit, target, size_average=False)
    factor = (without_size_average / with_size_average).data[0]
    print("- {loss.__name__}: {factor}".format(**locals()))```
```

```
Torch version: 0.2.0+40ca356

How each loss function scales the result with size_average:
- kl: 15.0
- mse: 15.0
- xe: 3.0
```

**Comment by zou3519** [comment]
A lot of the other loss functions behave like mse_loss: size_average=True means normalizing by dividing by total number of elements. I agree the documentation could be clearer, though: thanks for the report @tarvaina

**Comment by tarvaina** [comment]
Thanks for the answer. Yes, if size_average=True always means dividing by the total number of the elements, I guess it's best to keep it like that, and point it out clearly in the documentation of that parameter.

It is quite surprising behavior in the case of **mean** squared error, since the parameter makes it a sum instead of a mean.

**Comment by zou3519** [comment]
Are you saying that in the case where `size_average=False`, MSELoss behaves as a sum rather than as a mean?

**Comment by tarvaina** [comment]
Yes.

```python
from torch import zeros, ones
from torch.nn.functional import mse_loss

print(mse_loss(zeros(2, 2), ones(2, 2), size_average=True).data[0],
      mse_loss(zeros(2, 2), ones(2, 2), size_average=False).data[0])
```

outputs

```
1.0 4.0
```

**Comment by soumith** [comment]
this is expected behavior, we have to document it better. `size_average` is not just a setting for the mini-batch, but for "every atomic element where the loss is being calculated". The behavior has also been grandfathered in from torch5 (that's 3 torch versions out)

**Comment by tarvaina** [comment]
Sounds reasonable to me.
